libs/pulumi-lambda/src/lib/metric-alarms.ts

```diff
@@ -0,0 +1,289 @@
+             */
+            snsTopicArn: pulumi.Input<string>
+            /**
+             * The name of the function to monitor
+             */
+            lambdaFunctionName: pulumi.Input<string>
+            /**
+             * Set custom thresholds for the alarms
+             */
+            thresholds?: Thresholds
+        },
+        opts?: pulumi.ComponentResourceOptions,
+    ) {
+        super('wanews:lambda/MetricAlarms', name, {}, opts)
+
+        new aws.cloudwatch.MetricAlarm(
+            `${name}-err-rate`,
+            {
+                metricQueries: [
+                    {
+                        id: 'errorRate',
+                        label: 'Error rate (%)',
+                        expression: '100 * errors / MAX([errors, invocations])',
+                        returnData: true,
+                    },
+                    {
+                        id: 'errors',
+                        label: 'Errors',
+                        metric: {
+                            namespace: 'AWS/Lambda',
+                            metricName: 'Errors',
+                            stat: 'Sum',
+                            dimensions: {
+                                FunctionName: args.lambdaFunctionName,
+                                Resource: args.lambdaFunctionName,
+                            },
+                            period: 60,
+                        },
+                    },
+                    {
+                        id: 'invocations',
+                        label: 'Invocations',
+                        metric: {
+                            namespace: 'AWS/Lambda',
+                            metricName: 'Invocations',
+                            stat: 'Sum',
+                            dimensions: {
+                                FunctionName: args.lambdaFunctionName,
+                                Resource: args.lambdaFunctionName,
+                            },
+                            period: 60,
+                        },
+                    },
+                ],
+                evaluationPeriods: 2,
+                datapointsToAlarm: 2,
+                comparisonOperator: 'GreaterThanOrEqualToThreshold',
+                threshold:
+                    args.thresholds?.errorRatePercent ??
+                    defaultThresholds.errorRatePercent,
+                alarmDescription: pulumi.interpolate`threshold: error rate >= ${
+                    args.thresholds?.errorRatePercent ??
+                    defaultThresholds.errorRatePercent
+                }%: ${args.lambdaFunctionName}`,
+                alarmActions: [args.snsTopicArn],
+                okActions: [args.snsTopicArn],
+                insufficientDataActions: [args.snsTopicArn],
+                treatMissingData: 'notBreaching',
+            },
+            {
+                parent: this,
+                deleteBeforeReplace: true,
+            },
+        )
+
+        new aws.cloudwatch.MetricAlarm(
```

Comment:
Do we want to report on this, or it's more something we want to see on a dashboard?